modules/appeals_api/app/services/appeals_api/uploads/document_validation.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'pdf_info'
+
+module AppealsApi
+  module Uploads
+    class DocumentValidation
+      def initialize(upload)
+        @document = upload[:document]
+      end
+
+      def validate
+        # TODO: log anything here if errors raised or log in the controller if fails validation?
+        raise UploadValidationError, non_pdf_error unless document_is_pdf?
+        raise UploadValidationError, max_dimension_error unless valid_page_dimensions?
+        raise UploadValidationError, file_size_error unless valid_file_size?
+      end
+
+      def document_is_pdf?
+        contents = File.read(@document)
+        contents.include?('%PDF-')
```

Comment:
How much of the file needs to be read to determine this? Right now it appears to read the entire file into memory to check for any mention of `%PDF-` anywhere in the file, which could be a problem for large files.

Assuming this is something that can be found in the header of the file, we should try to only read a minimum amount of data before breaking away and failing.

Reply:
OK. Cool. I need to research a bit more, because it's possible it's always captured within the first 8 characters, but I haven't been able to find anything absolutely definitive. I'll keep looking. I'll let you know what I find.